src/periphery/LiquidationHelper.sol

```diff
@@ -0,0 +1,119 @@
+    function liquidateMaxTo(
+        address cauldron,
+        address account,
+        address recipient,
+        uint8 cauldronVersion
+    ) public returns (uint256 collateralAmount, uint256 borrowAmount) {
+        uint256 borrowPart = ICauldronV2(cauldron).userBorrowPart(account);
+        return liquidateTo(cauldron, account, recipient, borrowPart, cauldronVersion);
+    }
+
+    /// @notice Liquidate an account using a part of the borrow amount
+    function liquidate(
+        address cauldron,
+        address account,
+        uint256 borrowPart,
+        uint8 cauldronVersion
+    ) external returns (uint256 collateralAmount, uint256 borrowAmount) {
+        return liquidateTo(cauldron, account, msg.sender, borrowPart, cauldronVersion);
+    }
+
+    /// @notice Liquidate an account using a part of the borrow amount and send the collateral to a different address
+    function liquidateTo(
+        address cauldron,
+        address account,
+        address recipient,
+        uint256 borrowPart,
+        uint8 cauldronVersion
+    ) public returns (uint256 collateralAmount, uint256 borrowAmount) {
+        (collateralAmount, borrowAmount) = CauldronLib.getLiquidationCollateralAndBorrowAmount(ICauldronV2(cauldron), borrowPart);
+
+        IBentoBoxV1 box = IBentoBoxV1(ICauldronV2(cauldron).bentoBox());
+        uint256 shareMIMBefore = _transferRequiredMiMToCauldronDegenBox(box, borrowAmount);
+
+        IERC20 collateral = ICauldronV2(cauldron).collateral();
+        address masterContract = address(ICauldronV2(cauldron).masterContract());
+        box.setMasterContractApproval(address(this), masterContract, true, 0, 0, 0);
+
+        address[] memory users = new address[](1);
+        users[0] = account;
+        uint256[] memory maxBorrowParts = new uint256[](1);
+        maxBorrowParts[0] = borrowPart;
+
+        if (cauldronVersion <= 2) {
+            ICauldronV2(cauldron).liquidate(users, maxBorrowParts, address(this), address(0));
+        } else if (cauldronVersion == 3) {
+            ICauldronV3(cauldron).liquidate(users, maxBorrowParts, address(this), address(0), new bytes(0));
+        } else if (cauldronVersion == 4) {
+            ICauldronV4(cauldron).liquidate(users, maxBorrowParts, address(this), address(0), new bytes(0));
+        } else {
+            revert ErrInvalidCauldronVersion(cauldronVersion);
+        }
+
+        box.setMasterContractApproval(address(this), masterContract, false, 0, 0, 0);
+
+        // withdraw/refund any MiM left and withdraw collateral to wallet
+        box.withdraw(mim, address(this), msg.sender, 0, box.balanceOf(mim, address(this)) - shareMIMBefore);
+        box.withdraw(collateral, address(this), recipient, 0, box.balanceOf(collateral, address(this)));
+    }
+
+    /// @notice Transfer MiM from the liquidator to the BentoBox
+    function _transferRequiredMiMToCauldronDegenBox(IBentoBoxV1 box, uint256 amount) internal returns (uint256 shareMIMBefore) {
```

Comment:
Is the capitalization of MiM intentional? Should we just do `_transferRequiredMiMToCauldronDegenBox` instead?

Reply:
Just no-so-good ol' camel casing habbit. MIM it should be.